### PR TITLE
Make mpich default backend for unidist-mpi

### PR DESCRIPTION
**Pull request by YarShev**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,7 +12,7 @@ source:
 
 build:
   noarch: python
```
Comment (YarShev) [inline recipe/meta.yaml]:
```suggestion
```

**Comment by YarShev** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
 mpi:
```
Comment (dalcinl) [inline recipe/conda_build_config.yaml:1]: Look [here](https://github.com/conda-forge/mpi4py-feedstock/blob/main/recipe/conda_build_config.yaml). You need something like

```yaml
mpi:
  - mpich    # [not win]
  - openmpi  # [not win]
  - msmpi    # [win]
```

**Comment by YarShev** [comment]
Closing this PR. See the details in https://github.com/conda-forge/unidist-packages-feedstock/pull/11.

**Pull request closed without merging**